pkg/linkValidator/linkValidator.go

```diff
@@ -0,0 +1,82 @@
+package linkValidator
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	Timeout = time.Second * 5
+)
+
+// LinkValidator handles validation of URLs in service log messages
+type LinkValidator struct {
+	timeout    time.Duration
+	httpClient *http.Client
+}
+
+// ValidationResult holds a URL validation warning for non-fatal HTTP errors
+type ValidationResult struct {
+	URL     string
+	Warning error
+}
+
+// NewLinkValidator creates a new LinkValidator with default settings
+func NewLinkValidator() *LinkValidator {
+	return &LinkValidator{
+		timeout:    Timeout,
+		httpClient: &http.Client{Timeout: Timeout},
+	}
+}
+
+// Extract URLs from service log
+func extractURLs(text string) []string {
```

Comment:
No need to comment private functions, but if you still want to, the format should be `// extractURLs <text>`